test/serializers/associations_test.rb

```diff
@@ -145,6 +145,15 @@ def test_associations_custom_keys
         assert expected_association_keys.include? :writer
         assert expected_association_keys.include? :site
       end
+
+      def test_associated_and_nested_predicates
+        @post_serializer.each_association do |key, association|
+          if key == :blog
+            assert association.associated?
+            assert association.nested?
+          end
+        end
+      end
```

Comment:
would be good to have a sanity check assertion that the assertion were called. e.g. before the block set `called=false` and after the assertions set `called=true` then after the block `assert called`